text/0000-pack-buildpack-create.md

```diff
@@ -0,0 +1,105 @@
+# Meta
+[meta]: #meta
+- Name: Pack Command to Create a Buildpack Repo
+- Start Date: 2021-01-19
+- Author(s): [jkutner](https://github.com/jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal for a new command in Pack that would create the scaffolding for a new buildpack.
+
+# Definitions
+[definitions]: #definitions
+
+- *Scaffold* - the essential files and directories required for a buildpack
+
+# Motivation
+[motivation]: #motivation
+
+Every new buildpack requires a few essential files and directories. We can make buildpack development easier by automatically generating these with a single command.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Add a new command to Pack:
+
+```
+$ pack buildpack new <id> \
+    --api <api> \
+    --name <name> \
+    --path <path> \
+    --stack <stack> \
+    --version <version>
+```
+
+Where the flags include:
+
+* `api` - the buildpack API compatibility version
+* `name` - the name of the buildpack in the `buildpack.toml`
+* `path` - the location on the filesystem to generate the artifacts
+* `stack` - a set of compatible stacks for the buildpack. may be specificied multiple times. (defaults to `io.buildpacks.stacks.bionic` because it is required)
+* `version` - the version of the buildpack in `buildpack.toml` (defaults to `0.0.0` because it is required)
+
+The command will create a directory named for the buildpack ID, or the second component of a `<namespace>/<name>` ID. Inside of that directory, it will generate the minimal artifacts required to implement a buildpack in Bash.
+
+The `pack buildpack new` is only a generator. It will not be a _living command_ (i.e. you cannot re-run it to update a buildpack).
+
+## Future Work
+
+In the future, we may add support for a `--template` flag, which would create buildpacks from templates stored in a curated repository. In this way we can support many different languages, and even support versions the same language with different opinions.
```

Comment:
This is very interesting! An alternative to consider would be having a functionality similar to [cookiecutter](https://github.com/cookiecutter/cookiecutter) where you can provide a URI to a template (either a VCS uri or on the local fs). This would make it easier to handle new template contributions and they don't need to be put in a single repository (which may be a bottleneck in terms of reviews/flexibility?).